reviews/dbsc-analysis.md

```diff
@@ -0,0 +1,158 @@
+
+We have an alternative below that doesn't require an interactive exchange (though it could be made interactive).  It also includes a redundant new field in requests that lists a session identifier. That new field could be replaced either with a per-account resource URL parameter or a non-DBSC cookie.
+
+This process addresses an important concern about the frequency with which the browser needs to access the secret key.  Sites are able to control how often something is signed by setting the expiration date of the cookies they produce. The browser only needs to use the secret key when cookies expire.  Expirations are naturally limited because servers are unable to set extremely short expiration times without risking the cookies being completely useless to clients with bad clock skew; the granularity of expiration dates for cookies is also extremely limited in expressiveness.
+
+That uncertainty is another problem with the design. Due to that clock skew on clients, servers cannot be sure when clients will expire their cookies and initiate the update process.
+
+## Enrollment
+
+The design proposes the use of a new HTTP header field. If a server ever sent this field, in any HTTP response, that would initiate the enrollment process.
+
+This new field identifies the session resource. It also includes a challenge that needs to be signed to complete enrollment[^1]. The server also lists the types of keys it supports.
+
+To complete enrollment, the browser needs to generate a key pair, sign the challenge, and post the signature to the session resource. The content of that response is a JSON document that describes the rest of the protocol: which resources and cookies are governed by this resource.
+
+It’s not clear what it means to have multiple sessions concurrently active. There’s some mention of this, but it’s not fully developed. At least theoretically, there is no need to limit the number of sessions, other than to limit the work that browser and server both do and to reduce the state they both need to track.
+
+## Too Complex
+
+Overall, this creates a new set of interaction paradigms between the browser and websites.  We think that there are easier ways to achieve the same basic goals without too much disruption to the existing cookie handling arrangements. That design is sketched below.
+
+# An Alternative Design
+
+This is a sketch of an alternative approach that is far closer to how the web platform currently handles cookies.
+
+The core requirement for the usage part is that the site is able to regularly request that the browser demonstrate that it has access to the private key that was registered through the enrollment process.  Ideally, that access is not required for most interactions, because generating and validating a digital signature is somewhat expensive for both client and server.
```

Comment:
This contradicts footnote [^2], which claims the client-side cost is trivial. We should be internally consistent.